cache.go

```diff
@@ -7,78 +7,79 @@ import (
 
 // Cache is a synchronised map of items that auto-expire once stale
 type Cache struct {
-	mutex sync.RWMutex
+	sync.RWMutex
 	ttl   time.Duration
 	items map[string]*Item
 }
 
+// NewCache is a helper to create instance of the Cache struct
+func New(duration time.Duration) *Cache {
+	cache := &Cache{
+		ttl:   duration,
+		items: map[string]*Item{},
+	}
+	cache.startEvictionTimer()
+	return cache
+}
+
 // Set is a thread-safe way to add new items to the map
-func (cache *Cache) Set(key string, data string) {
-	cache.mutex.Lock()
+func (cache *Cache) Set(key string, data []byte) {
+	cache.Lock()
+	defer cache.Unlock()
+
 	item := &Item{data: data}
 	item.touch(cache.ttl)
 	cache.items[key] = item
-	cache.mutex.Unlock()
 }
 
 // Get is a thread-safe way to lookup items
 // Every lookup, also touches the item, hence extending it's life
-func (cache *Cache) Get(key string) (data string, found bool) {
-	cache.mutex.Lock()
+func (cache *Cache) Get(key string) ([]byte, bool) {
+	cache.Lock()
+	defer cache.Unlock()
+
 	item, exists := cache.items[key]
 	if !exists || item.expired() {
-		data = ""
-		found = false
-	} else {
-		item.touch(cache.ttl)
-		data = item.data
-		found = true
+		return nil, false
```

Comment:
Just a thought...is it possible for an item to be expired but still in the cache? Would the `evict` function hold the map and make sure it was gone before anything else could access it?